src/Tools/dotnet-monitor/DiagnosticsMonitorCommandHandler.cs

```diff
@@ -87,7 +92,56 @@ public IWebHostBuilder CreateWebHostBuilder(IConsole console, string[] urls, str
                         services.ConfigureMetrics(context.Configuration);
                     }
                 })
-                .UseUrls(urls)
+                .ConfigureKestrel( (context, options) =>
+                {
+                    bool boundListeningPort = false;
+
+                    //Workaround for lack of default certificate. See https://github.com/dotnet/aspnetcore/issues/28120
+                    options.Configure(context.Configuration.GetSection("Kestrel")).Load();
+
+                    //By default, we bind to https for sensitive data (such as dumps and traces) and bind http for
+                    //non-sensitive data such as metrics. We may be missing a certificate for https binding. We want to continue with the
+                    //http binding in that scenario.
+                    foreach (BindingAddress url in urls.Select(BindingAddress.Parse))
+                    {
+                        Action<ListenOptions> configureListenOptions = (listenOptions) =>
+                        {
+                            if (url.Scheme.Equals(Uri.UriSchemeHttps, StringComparison.OrdinalIgnoreCase))
+                            {
+                                listenOptions.UseHttps();
+                            }
+                        };
+
+                        try
+                        {
+                            if (url.Host.Equals("localhost", StringComparison.OrdinalIgnoreCase))
+                            {
+                                options.ListenLocalhost(url.Port, configureListenOptions);
+                            }
+                            else if (IPAddress.TryParse(url.Host, out IPAddress ipAddress))
+                            {
+                                options.Listen(ipAddress, url.Port, configureListenOptions);
+                            }
+                            else
+                            {
+                                options.ListenAnyIP(url.Port, configureListenOptions);
```

Comment:
The two previous conditions check for localhost and an IP address. Is there any other that needs to be supported? A cursory look at the Listen* methods seem so show that UDS and DNS endpoints are supported, but those don't really make sense for dotnet-monitor, so maybe what you have is the complete set.

Reply:
As far as I know this is the complete list, and closely mimics the asp.net core pattern:
- Attempt localhost
- Attempt an ip like string
- Anything else means bind to any